core/lib/network-recorder.js

```diff
@@ -273,6 +273,18 @@ class NetworkRecorder extends RequestEventEmitter {
         candidates = documentCandidates;
       }
     }
+    if (candidates.length > 1) {
+      // If all real loads came from successful preloads (url preloaded and
+      // loads came from the cache), filter to link rel=preload request(s).
+      const linkPreloadCandidates = candidates.filter(c => c.isLinkPreload);
+      if (linkPreloadCandidates.length) {
+        const nonPreloadCandidates = candidates.filter(c => !c.isLinkPreload);
+        const allPreloaded = nonPreloadCandidates.every(c => c.fromDiskCache || c.fromMemoryCache);
+        if (nonPreloadCandidates.length && allPreloaded) {
```

Comment:
this does not handle the case of unsuccessful link rel preloads (e.g. `crossorigin` mismatch), but we could in the future.